### PR TITLE
Kitsu MAL Rating patch

### PR DESCRIPTION
**Pull request by synthtech**
Fixes some remaining issues:
- Certain MAL entries were structured differently which was causing errors
- The rating bar would be recreated each time when going to another media page
- The rating bar would not be inserted if the original was not on the page

### CODE REVIEW

**Comment by fuzetsu** [comment]
Looks good, but testing it out, I found this issue. Happens when switching to a page that doesn't have a kitsu rating bar from a page that has both. 

image

**Comment by fuzetsu** [comment]
I'll merge it because it's still better than the current live version, but would be good to sort out the timing issues. Sometimes if you go to another page and it swaps out the content too slowly the bar on the current page will get updated, and then removed when it finally swaps out the content.

File: kitsu-mal-rating/kitsu-mal-rating.user.js [hunk 1]
```diff
@@ -122,41 +130,69 @@
...
+            var updateRating = Util.q('.community-percentage', barCheck);
+            updateRating.textContent = rating;
+
+            var updateLink = Util.q('a', barCheck);
+            updateLink.href = malLink;
+
+            var updateUsers = Util.q('.ratings-count', barCheck);
+            updateUsers.textContent = usersRated + ' ratings - ' + usersFaved + ' favorites';
+
+            if (!ratingBar.hasAttribute('style')) {
```
Comment (fuzetsu) [inline kitsu-mal-rating/kitsu-mal-rating.user.js]: this line is where the error happens, content gets swapped out between `waitForUrl` firing and `getMalLink` and `barCheck` is out of date.

**Comment by synthtech** [comment]
Sorry about that. I think some recent changes with the way media pages are loaded sped things up making the bug much more apparent since my testing from a couple days ago didn't show anything unusual.

**Comment by fuzetsu** [comment]
Makes sense, and thanks again for the pulls.

**Pull request merged**